### DISCUSSION TITLE
Prefix configuration does not work

### DISCUSSION THREAD

**Discussion by lamualfa** [windicss/vite-plugin-windicss#77] [Q&A]
## Setup 1 (_Not Work_)

```js
windiCSS({
  config: {
    prefixer: true,
    prefix: 'lamualfa',
  },
});
```
<hr>

## Setup 2 (_Not Work_)

```js
windiCSS({
  prefixer: true,
  prefix: 'lamualfa',
});
```
<hr>

## Setup 3 (_Not Work_)

```js
windiCSS({
  compile: true,
  prefixer: true,
  prefix: 'lamualfa',
});
```

How do i setting the `prefix` config?

# Environment

- vite: `v2.0.5`
- vite-plugin-windicss: `v0.7.2`
- svelte: `v3.35.0`

# Reference

- https://github.com/windicss/vite-plugin-windicss/blob/main/packages/plugin-utils/src/options.ts
- https://github.com/windicss/windicss/blob/main/src/interfaces.ts

**Comment by hannoeru** [comment]
Can you provide more context?

**Reply by lamualfa** [reply]
I use this template [svelte-routify-windi-vite](https://github.com/reepolee/svelte-routify-windi-vite) with updated dependencies as described in **Environment** section and with configuration above.

**Comment by schelmo** [comment]
just tested this with
```js
windiCSS({
  prefixer: true,
  prefix: 'lamualfa',
});
```
your classes are now for example: lamualfatext-red-500
this works with the latest [svelte-routify-windi-vite](https://github.com/reepolee/svelte-routify-windi-vite) template
maybe you want a hypen (for better readability) ? ```prefix: 'lamualfa-'```

**Reply by hannoeru** [reply]
I think `prefixer` is default to `true` and not related to `prefix`, so you can remove it.

**Reply by lamualfa** [reply]
@schelmo  @hannoeru Ok. Thanks bro. I will check it now.

**Reply by lamualfa** [reply]
@schelmo I have been tested with the latest version of [svelte-routify-windi-vite](https://github.com/reepolee/svelte-routify-windi-vite), but it's still not work in build mode.

**The output of the CSS file in `dist` directory after i run `pnpm build` command.**

![image](https://user-images.githubusercontent.com/39755201/111550498-d146bd00-87b0-11eb-80bc-a67e81df5f10.png)
>`.py-4`, `.text-4xl`, `.text-center` and etc is not changed.

<hr>

## Here are some of the configurations I have tried

### Config 1 (_Not work_)

`vite.config.ts`

```js
      windiCSS({
        //@ts-ignore
        verbose: true,
        silent: false,
        debug: true,
        config: "tailwind.config.js", // tailwind config file path (optional)
        compile: true, // false: interpretation mode; true: compilation mode
        prefixer: true,
        prefix: "windi-", // set compilation mode style prefix
        globalPreflight: true, // set preflight style is global or scoped
        globalUtility: true, // set utility style is global or scoped
      }),
```
<hr>

### Config 2 (_Not work_)


`vite.config.ts`

```js
      windiCSS({
        //@ts-ignore
        verbose: true,
        silent: false,
        debug: true,
        config: "tailwind.config.js", // tailwind config file path (optional)
        compile: false, // false: interpretation mode; true: compilation mode
        prefixer: true,
        prefix: "windi-", // set compilation mode style prefix
        globalPreflight: true, // set preflight style is global or scoped
        globalUtility: true, // set utility style is global or scoped
      }),
```
<hr>

### Config 3 (_Not work_)

I try to move the config to the `tailwind.config.js` file

`tailwind.config.js`

```js
module.exports = {
  compile: true, // false: interpretation mode; true: compilation mode
  prefixer: true,
  prefix: 'windi-', // set compilation mode style prefix
  theme: {
    extend: {
      colors: {
        reepolee: {
          500: '#b40000',
          600: '#9c0000',
          700: '#750000',
        },
      },
    },
  },
};
```

<hr>

I use PNPM for the package manager. Does it matter?

**Reply by schelmo** [reply]
pnpm should be no problem.
Did you change the classes in your html to have the windi- prefix?
windicss is also generating the classes without prefix, if they are found in the html.
if you use a tailwind.config.js, you should put the prefix configuration there.
i made a commit here so you can compare:
https://github.com/schelmo/svelte-routify-windi-vite/commit/cdedd49acf5154d7b9c3ac5fb19a059598df91d4

**Comment by lamualfa** [comment]
@schelmo 

> Did you change the classes in your html to have the windi- prefix?

i want that. but i don't do it manually. i want Windi CSS to do it automatically in build mode.

<hr>

> if you use a `tailwind.config.js`, you should put the prefix configuration there.

I've tried it. but still not working.

<hr>

> i made a commit here so you can compare:

Ok, thanks. I will try it.
